designer/server/src/test/scala/pl/touk/nussknacker/ui/notifications/NotificationServiceTest.scala

```diff
@@ -316,37 +308,15 @@ class NotificationServiceTest
     )
     val deploymentService = new DeploymentService(
       managerDispatcher,
-      mock[ProcessingTypeDataProvider[UIProcessValidator, _]],
-      mock[ProcessingTypeDataProvider[ScenarioResolver, _]],
+      TestFactory.processValidatorByProcessingType,
+      TestFactory.scenarioResolverByProcessingType,
       actionService,
       TestFactory.additionalComponentConfigsByProcessingType,
     ) {
       override protected def validateBeforeDeploy(
           processDetails: ScenarioWithDetailsEntity[CanonicalProcess],
-          deployedScenarioData: DeployedScenarioData,
-          updateStrategy: DeploymentUpdateStrategy
+          runDeploymentCommand: DMRunDeploymentCommand,
       )(implicit user: LoggedUser): Future[Unit] = Future.successful(())
-
-      override protected def prepareDeployedScenarioData(
```

Comment:
I replaced this hack by passing processValidatorByProcessingType and scenarioResolverByProcessingType